elements/pfe-avatar/src/pfe-avatar.js

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright 2018 Red Hat, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import PFElement from "../pfelement/pfelement.js";
+import { hash } from "./djb-hash.js";
+import { hsl2rgb, rgb2hsl } from "./hslrgb.js";
+
+class PfeAvatar extends PFElement {
+  static get tag() {
+    return "pfe-avatar";
+  }
+
+  get templateUrl() {
+    return "pfe-avatar.html";
+  }
+
+  get styleUrl() {
+    return "pfe-avatar.scss";
+  }
+
+  static get observedAttributes() {
+    return ["name", "pattern", "src", "shape"];
+  }
+
+  static get patterns() {
+    return {
+      triangles: "triangles",
+      squares: "squares"
+    };
+  }
+
+  static get defaultColors() {
+    return "#3B0083 #f0ab00 #007a87 #00b9e4 #92d400";
+  }
+
+  get name() {
+    return this.getAttribute("name");
+  }
+
+  set name(val) {
+    return this.setAttribute("name", val);
+  }
+
+  get src() {
+    return this.getAttribute("src");
+  }
+
+  set src(href) {
+    return this.setAttribute("src", href);
+  }
+
+  get pattern() {
+    return this.getAttribute("pattern") || PfeAvatar.patterns.squares;
+  }
+
+  set pattern(name) {
+    if (!PfeAvatar.patterns[name]) {
+      this.log(
+        `invalid pattern "${name}", valid patterns are: ${Object.values(
+          PfeAvatar.patterns
+        )}`
+      );
+      return;
+    }
+    return this.setAttribute("pattern", name);
+  }
+
+  constructor() {
+    super(PfeAvatar);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this._initCanvas();
+
+    this.dispatchEvent(
+      new CustomEvent(`${PfeAvatar.tag}:connected`, {
+        bubbles: false
+      })
+    );
+  }
+
+  attributeChangedCallback(attr, oldValue, newValue) {
+    super.attributeChangedCallback(...arguments);
+
+    switch (attr) {
```

Comment:
If these all break without running anything, what's the point of this switch statement?

Reply:
Just a placeholder I forgot to remove.